File: dotnet-desktop-guide/net/wpf/windows/snippets/dialog-boxes-overview/csharp/MarginValidationRule.cs
﻿using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Controls;

namespace Dialogs
{
    public class MarginValidationRule : ValidationRule
    {
        public double MinMargin { get; set; }
        public double MaxMargin { get; set; }

        public override ValidationResult Validate(object value, CultureInfo cultureInfo)
        {
            double margin;

            // Is a number?
            if (!double.TryParse((string)value, out margin))
            {
                return new ValidationResult(false, "Not a number.");
            }

            // Is in range?
            if ((margin < MinMargin) || (margin > MaxMargin))
            {
                var msg = $"Margin must be between {MinMargin} and {MaxMargin}.";
                return new ValidationResult(false, msg);
            }

            // Number is valid
            return new ValidationResult(true, null);
        }
    }
}

File: dotnet-desktop-guide/net/wpf/windows/snippets/how-to-close-window-dialog-box/vb/MarginValidationRule.vb
Imports System.Globalization

Public Class MarginValidationRule
    Inherits ValidationRule

    Public Property MinMargin As Double
    Public Property MaxMargin As Double

    Public Overrides Function Validate(value As Object, cultureInfo As CultureInfo) As ValidationResult
        Dim margin As Double

        ' Is it a number?
        If Not Double.TryParse(DirectCast(value, String), margin) Then
            Return New ValidationResult(False, "Not a number")
        End If

        ' Is it in range
        If margin < MinMargin Or margin > MaxMargin Then
            Return New ValidationResult(False, $"Margin must be between {MinMargin} and {MaxMargin}")
        End If

        Return New ValidationResult(True, Nothing)

    End Function

End Class
